src/asm2cfg/asm2cfg.py

```diff
@@ -191,48 +192,109 @@ def __str__(self):
         return ' '.join(map(lambda b: f'{b:#x}', self.bites))
 
 
+class X86TargetInfo:
+    """
+    Contains instruction info for X86-compatible targets.
+    """
+
+    def __init__(self):
+        pass
+
+    def comment(self):
+        return '#'
+
+    def is_call(self, instruction):
+        # Various flavors of call:
+        #   call   *0x26a16(%rip)
+        #   call   0x555555555542
+        #   addr32 call 0x55555558add0
+        return 'call' in instruction.opcode
+
+    def is_jump(self, instruction):
+        return instruction.opcode[0] == 'j'
+
+    def is_unconditional_jump(self, instruction):
+        return instruction.opcode.startswith('jmp')
+
+    def is_sink(self, instruction):
+        return instruction.opcode.startswith('ret')
+
+
+class ARMTargetInfo:
+    """
+    Contains instruction info for X86-compatible targets.
+    """
+
+    def __init__(self):
+        pass
+
+    def comment(self):
+        return ';'
+
+    def is_call(self, instruction):
+        # Various flavors of call:
+        #   bl 0x19d90 <_IO_vtable_check>
+        # Note that we should be careful to not mix it with conditional
+        # branches like 'ble'.
+        return instruction.opcode.startswith('bl') \
+            and instruction.opcode not in ('blt', 'ble', 'bls')
+
+    def is_jump(self, instruction):
+        return instruction.opcode[0] == 'b' and not self.is_call(instruction)
+
+    def is_unconditional_jump(self, instruction):
+        return instruction.opcode == 'b'
+
+    def is_sink(self, instruction):
+        # Detect various flavors of return like
+        #   bx lr
+        #   pop {r2-r6,pc}
+        # Note that we do not consider conditional branches (e.g. 'bxle') to sink.
+        return re.search(r'\bpop\b.*\bpc\b', instruction.body) \
+          or (instruction.opcode == 'bx' and instruction.ops[0] == 'lr') \
+          or instruction.opcode == 'udf'
+
+
 class Instruction:
     """
     Represents a single assembly instruction with it operands, location and
     optional branch target
     """
-    def __init__(self, body, text, lineno, address, opcode, ops, target, imm):  # pylint: disable=too-many-arguments
+    def __init__(self, body, text, lineno, address, opcode, ops, target, imm, target_info):  # noqa
         self.body = body
         self.text = text
         self.lineno = lineno
         self.address = address
         self.opcode = opcode
         self.ops = ops
         self.target = target
+        self.info = target_info
         if imm is not None and (self.is_jump() or self.is_call()):
             if self.target is None:
                 self.target = imm
             else:
                 self.target.merge(imm)
 
+    def __str__(self):
```

Comment:
I would prefer to move methods like __str__ to last in the class because they are only used to print the object and are not part of the core functionality.